src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashAlgorithmNames.cs

```diff
@@ -63,6 +68,21 @@ public static string ToUpper(string hashAlgorithmName)
                 return SHA1;
             }
 
+            if (hashAlgorithmName.Equals(SHA3_256, StringComparison.OrdinalIgnoreCase))
```

Comment:
I left out modifying `ToUpper` intentionally because as far as I was able [to observe](https://github.com/search?q=repo%3Adotnet%2Fruntime%20%2FHashAlgorithmNames.ToUpper%2F&type=code) it's only used in `RSAPKCS1Signature{De}formatter`. These types are only relevant in System.Security.Cryptography.Xml and CryptoConfig. CryptoConfig which isn't getting SHA3 support yet, because Windows didn't add any SHA3 OIDs to their OID registry.

The same goes for `ToAlgorithmName`.

If we can write some tests that show that adding these here matters in a place where we want to support SHA-3, that would be helpful.

Reply:
Should have known it was intentional, sorry. I did write some tests and [pushed them here](https://github.com/dotnet/runtime/pull/87237/commits/b229b0db967f0f4592eef13c8e7a623969cee445) however I couldn't actual compile them as `HashAlgorithmNames.cs` is internal. Do we HAVE a path/process for doing tests on those (just coming back to .Net after many years stale, sorry)?

Reply:
> Should have known it was intentional, sorry.

All good. I'm really happy someone is looking through the changes!

To set the expectation though, there are likely going to be many places you will encounter where we handling SHA1... SHA2... MD5... but it looks like we "forgot" about SHA3. A good number of those places are going to be intentional - Windows hasn't wired SHA3 in to its whole stack yet (like X.509 or EncryptedXml, etc). So we continue to operate as we always have, especially where we want exception-for-exception compatibility with .NET Framework.

>  Do we HAVE a path/process for doing tests on those (just coming back to .Net after many years stale, sorry)?

Generally, no. The public API surface is what gets tested. There are some exceptions to this though, where we use reflection to peek at non-public members, but try to do so rarely. Things like InternalsVisibleTo are not used and discouraged, last time as I understood it.

Reply:
Cool, thanks for the education. I'll close this PR now :)